app/Models/Competition.php

```diff
@@ -33,6 +33,14 @@ public function subscribers()
         return $this->belongsToMany(User::class, 'competition_user', 'competition_id', 'northstar_id')->wherePivot('unsubscribed', '=', 0);
     }
 
+    public function unsubscribeUser($competition_id, $northstar_id)
```

Comment:
You don't need to define the relationship again with `belongsToMany` since it has already been defined above. Also once you are in the `Competition` model you are already aware of the `id`and the `user`  (since they are properties/relationships on that instance of the model) so you shouldn't need to pass it here.

All you need to do is grab the record from the pivot table and update that 
This method could look more like this:
```
public function unsubscribe() {
  return DB::table('competitions_users')->where([
      ['northstar_id', '=', $this->user ],
      ['competition', '=', $this->id],
  ])->update(['unsubscribed' => 1]);
}
```